pkg/cmd/search/repos/repos.go

```diff
@@ -16,66 +14,80 @@ import (
 	"github.com/spf13/cobra"
 )
 
+const (
+	// Limitation of GitHub search see:
+	// https://docs.github.com/en/rest/reference/search
+	searchMaxResults = 1000
+)
+
 type ReposOptions struct {
-	Browser    cmdutil.Browser
-	Config     func() (config.Config, error)
-	Exporter   cmdutil.Exporter
-	HttpClient func() (*http.Client, error)
-	IO         *iostreams.IOStreams
-	Query      search.Query
-	Searcher   search.Searcher
-	WebMode    bool
+	Browser  cmdutil.Browser
+	Exporter cmdutil.Exporter
+	IO       *iostreams.IOStreams
+	Query    search.Query
+	Searcher search.Searcher
+	WebMode  bool
 }
 
 func NewCmdRepos(f *cmdutil.Factory, runF func(*ReposOptions) error) *cobra.Command {
+	var order string
+	var sort string
 	opts := &ReposOptions{
-		Browser:    f.Browser,
-		Config:     f.Config,
-		HttpClient: f.HttpClient,
-		IO:         f.IOStreams,
-		Query:      search.Query{Kind: search.KindRepositories},
+		Browser: f.Browser,
+		IO:      f.IOStreams,
+		Query:   search.Query{Kind: search.KindRepositories},
 	}
 
 	cmd := &cobra.Command{
 		Use:   "repos [<query>]",
 		Short: "Search for repositories",
 		Long: heredoc.Doc(`
-      Search for repositories on GitHub.
+			Search for repositories on GitHub.
+
+			The command supports constructing queries using the GitHub search syntax,
+			using the parameter and qualifier flags, or a combination of the two.
 
-      The command supports constructing queries using the GitHub search syntax,
-      using the parameter and qualifier flags, or a combination of the two.
+			GitHub search syntax is documented at:
+			https://docs.github.com/search-github/searching-on-github/searching-for-repositories
     `),
 		Example: heredoc.Doc(`
-      # search repositories matching set of keywords "cli" and "golang"
-      $ gh search repos cli golang
+			# search repositories matching set of keywords "cli" and "shell"
+			$ gh search repos cli shell
 
-      # search repositories matching phrase "cli golang"
-      $ gh search repos "cli golang"
+			# search repositories matching phrase "vim plugin"
+			$ gh search repos "vim plugin"
 
-      # search repositories in the cli organization
-      $ gh search repos --org=cli
+			# search repositories public repos in the microsoft organization
+			$ gh search repos --owner=microsoft --visibility=public
 
-      # search repositories with a set of topics
-      $ gh search repos --topic="cli, terminal"
+			# search repositories with a set of topics
+			$ gh search repos --topic="unix, terminal"
```

Comment:
💅 Let's not use a space here, even though it doesn't hurt, to avoid training users to use spaces in comma-separated lists, and to allow them to not have to put quotes around this